static/docs/user-guide/index.md

```diff
@@ -27,6 +27,9 @@ guides related to contributing to this
   [DVC-file](/doc/user-guide/dvc-files-and-directories) dependencies.
 - [Managing External Data](/doc/user-guide/managing-external-data) explains both
   external outputs and external cache.
+- [Writing well-behaved commands](/doc/user-guide/well-behaved-command) goes
+  through the few technicalities that you need to take into account when using
+  DVC to run your commands.
```

Comment:
Agree with Ivan. No need for a whole new guide for this note, let's just add it in the **Description** section of the `dvc run` cmd ref (It can be a paragraph or even an H3 mini section. But maybe also a shorter note in other relevant places where `dvc run `is introduced? E.g. Get Started or some Tutorials